### ISSUE TITLE
Two suggestions for log filtering

### ISSUE THREAD

**Issue by tflo** [ketho-wow/GuildBankSearch#2]
The bank search/filter works perfectly, but for **log** filtering I have two suggestions:

- Match the search string against the entire log text, not only against the item names:
  - This would allow to search also for player names, to quickly see for example when and in which tab I or someone has deposited/withdrawn stuff.

- Add some sort of indication in which tab’s log the term was found:
  - Unlike the bank search, if a search term is contained in the log, the tab is not highlighted. That means – unless I’m missing something – I have to click through all the tabs in order to actually see if a search term was found in the log anywhere.
  - This could be indicated by a small overlay symbol over the tab icon (for example).
    - Tab icon highlighted = found in bank; Log symbol on icon = found in log; both = found in both

Thanks for considering, and thanks for this useful addon

– Tom

**Comment by Ketho** [comment]
- That sound like a good idea. I will at least implement searching for player names in the log text.

![image](https://user-images.githubusercontent.com/1073877/193487551-8c35c09d-537a-4164-981a-1a05b4a2bd48.png)

- About highlighting the tab when a search term is contained in the log for items (and players). It might be confusing, I suppose normally a user would only expect to search for items actually in the guild bank, not ones that are still lingering in the transaction log. Otherwise I agree highlighting the tab and adding a small overlay would be a nice solution.

Luckily I still had a test guild on the PTR and noticed the addon also needs some fixes for patch 10.0
edit: bit of misclicking when I added the label

**Comment by tflo** [comment]
> About highlighting the tab when a search term is contained in the log for items (and players). It might be confusing,

No, I meant a small overlay on the tab icon if found in the log, not highlighting. Highlighting should be reserved for a bank (item) match. So, highlight + overlay would mean it was found in both.

However, it occurs to me now that it might be very likely that an item in the bank will also be found in the log, so the overlay indicator really should not be too big.

Or simply adding a switch? An ‘Include Log’ checkbox. Just a thought…

**Comment by Ketho** [comment]
So just an overlay for items/names found in the log then. I'll try looking at it this week.

**Comment by tflo** [comment]
PS: 

The switch could also be an OR switch (instead of Include). As long as the search term gets not erased from the field, it would be easy enough to switch from bank to log search with just one click.

This would have the advantage that you don’t need a new indicator. The normal highlighting would serve for both then (and with the switch it would be clear what it means).

**Comment by Ketho** [comment]
I'm kind of afraid to add a checkbox option / switch since the addon doesn't have any yet, nor any savedvariables. I'm not good with design but just adding an overlay sounds more simple than finding a place for a checkbox somewhere <.<

edit: never mind my last question, I mistakenly thought I split the files for retail and classic wrath

**Comment by tflo** [comment]
> I'm kind of afraid to add a checkbox

OK :-)

Any solution will be fine, ofc! So back to my initial points (search for guildie names, and any sort of hint in which tab’s log the search term was found). Forget the rest, it was just loud thinking ;)

**Comment by Ketho** [comment]
I forgot to ask, is this for retail or classic wrath?

**Comment by tflo** [comment]
Sorry, haven’t noticed that you have versions for Classic too. I’m mainly playing Retail.

**Comment by Ketho** [comment]
I have been terribly busy lately, the overlay icon thing is still on my mind but I tagged a new release. I don't like to keep people waiting until it's done...

**Comment by tflo** [comment]
No worries, actually I don't feel like being waiting. Searching the guild bank or the logs is not something I do every day ;)
